src/editor/serialize.ts

```diff
@@ -61,9 +61,9 @@ export function htmlSerializeIfNeeded(model: EditorModel, {forceHTML = false} =
                 // const inlinePattern = "(?:^|\\s)(?<!\\\\)\\$(?!\\s)(([^$]|\\\\\\$)+?)(?<!\\\\|\\s)\\$";
 
                 // conditions for display math detection $$...$$:
-                // - pattern starts at beginning of line or is not prefixed with backslash or dollar
+                // - pattern starts and ends on a new line
                 // - left delimiter ($$) is not escaped by backslash
-                "display": "(^|[^\\\\$])\\$\\$(([^$]|\\\\\\$)+?)\\$\\$",
+                "display": "(^)\\$\\$(([^$]|\\\\\\$)+?)\\$\\$$",
```

Comment:
I think `(^)` can just become `^`?  (same with the other regexp)

Reply:
No, I pretty sure the regex replace code depends on this group there

Reply:
It also still allows someone to, for example, use `"display": "(^|[^\\\\$])\\$\\$(([^$]|\\\\\\$)+?)\\$\\$"` in their config

Reply:
Ah yes, I see now.  Looks good then.